pos_barcode_tare/models/pos_config.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+    iface_tare_label = fields.Boolean(
+        'Show tare label button',
+        help="Print tare labels with this POS"
+    )
+
+    iface_tare_barcode_sequence_id = fields.Integer(
```

Comment:
Hi. I don't think it is the good way to do. (mainly because you hardcode 36 ID in the code !)

I think that a more modular approach is (mainly) : 

* to create a new **type** of barcode rule (ref : https://github.com/odoo/odoo/blob/12.0/addons/barcodes/models/barcodes.py#L184) named ``tare``
* create a new function barcode_tare_action like (per exemple) ``barcode_cashier_action`` (https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/static/src/js/screens.js#L76)
* add a new action of BarcodeReader (ref) https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/static/src/js/devices.js#L501

Note : regarding your data, I don't think that 21 is a good prefix as it is reserved for other rule in Odoo "weightable products". See : https://github.com/odoo/odoo/blob/12.0/addons/stock/data/default_barcode_patterns.xml#L10
This rule should be used for weightable product, when .... = little internal product code and NNDDD is the weight.

for that purpose, I think that a new type like this is the good solution : 
```
    <data noupdate="1">
         <record id="barcode_rule_tare model="barcode.rule">
            <field name="name">Tare</field>
            <field name="barcode_nomenclature_id" ref="barcodes.default_barcode_nomenclature"/>
            <field name="sequence">DONT_CARE</field>
            <field name="type">tare</field>
            <field name="encoding">ean13</field>
            <field name="pattern">XX00000{NNDDD}</field>
        </record>
    </data>
```
Where XX should not be in existing values.

kind regards.

Reply:
Thank you for the detailed explanation! I'll do that,
Cheers.

Reply:
@legalsylvain this is done (re-using some of your code from `pos_price_to_weight`) as this is developed for v9.0.